test/test.externalclient.ts

```diff
@@ -116,6 +129,57 @@ describe('ExternalAccountClient', () => {
       );
     });
 
+    it('should return IdentityPoolClient with expected workforce configs', () => {
+      const validWorkforceIdentityPoolClientAudiences = [
+        '//iam.googleapis.com/projects/123/locations/global/workforcePools/workforcePools/providers/provider',
+        '//iam.googleapis.com/projects/workforcePool/locations/global/workforcePools/pool/providers/provider',
+        '//iam.googleapis.com/projects/projectId/locations/global/workforcePools/workloadPools/providers/oidc',
+      ];
+      const workforceFileSourcedOptions = Object.assign(
+        {},
+        fileSourcedOptions as any
+      );
+      workforceFileSourcedOptions.workforce_pool_user_project =
+        'work_force_pool_user_project';
+      workforceFileSourcedOptions.subject_token_type =
+        'urn:ietf:params:oauth:token-type:id_token';
+      for (const validWorkforceIdentityPoolClientAudience of validWorkforceIdentityPoolClientAudiences) {
+        workforceFileSourcedOptions.audience =
+          validWorkforceIdentityPoolClientAudience;
+        const expectedClient = new IdentityPoolClient(
+          workforceFileSourcedOptions
+        );
+
+        assert.deepStrictEqual(
+          ExternalAccountClient.fromJSON(workforceFileSourcedOptions),
+          expectedClient
+        );
+      }
+    });
+
+    invalidWorkforceIdentityPoolClientAudiences.forEach(
+      invalidWorkforceIdentityPoolClientAudience => {
+        const workforceIdentityPoolClientInvalidOptions = Object.assign(
+          {},
+          fileSourcedOptions as any
+        );
+        workforceIdentityPoolClientInvalidOptions.workforce_pool_user_project =
+          'work_force_pool_user_project';
+        workforceIdentityPoolClientInvalidOptions.subject_token_type =
+          'urn:ietf:params:oauth:token-type:id_token';
+        it(`should throw given audience ${invalidWorkforceIdentityPoolClientAudience} with user project defined in IdentityPoolClientOptions`, () => {
+          workforceIdentityPoolClientInvalidOptions.audience =
+            invalidWorkforceIdentityPoolClientAudience;
+
+          assert.throws(() => {
+            return ExternalAccountClient.fromJSON(
+              workforceIdentityPoolClientInvalidOptions
+            );
+          });
+        });
+      }
+    );
```

Comment:
Let's move the loop inside the test, it'll be much cleaner:

```
it('should throw an error when an invalid workforce audience is provided with a workforce user project', () => {
  for (...)
    const workforceIdentityPoolClientInvalidOptions = Object.assign(
      {},
      fileSourcedOptions as any, 
      {audience: invalidWorkforceIdentityPoolClientAudience, workforce_pool_user_project: workforce_pool_user_project}
    );

    assert.throws(() => {
      return ExternalAccountClient.fromJSON(
        workforceIdentityPoolClientInvalidOptions
      );
    });
});
```

Reply:
https://github.com/googleapis/google-auth-library-nodejs/pull/1229#discussion_r688270490